### ISSUE TITLE
More "ergonomic" express middleware for Cloud Functions

### ISSUE THREAD

**Issue by jacobg** [googleapis/nodejs-logging-winston#367]
A Cloud Function has a couple particular needs for which this Winston plugin can be improved:

1. A Cloud Function needs to synchronously export an Express application like this:
```
import * as functions from 'firebase-functions'
import * as express from 'express'
const app = express()
// *** configure the express app ***
// You can't do this:  await lw.express.middleware();
module.exports = functions.https.onRequest(app)
```
The only way I see initializing the winston middleware is to do some complicated injection into another a wrapper middleware function. It needs to be more easily usable, with a documented code example.

2. In local development, logging should just go to the console, not Stackdriver. But the middleware hardcodes the LoggingWinston transport that uploads logs to GCP. It seems that with the version 2.0 design that allows the user to pass in a logger to the middleware, it should also allow the user to decide which transports to use.

Related issues that I think were closed without resolution:
#355 
#303

**Comment by ofrobots** [comment]
Hi @jacobg . Thanks for the issue. This middleware API in this library in its current form is not yes the ideal choice for Google Cloud Functions for a bunch reasons:

1. As you have observed, the GCF programming model is that the serving environment is responsible for starting the express server and any necessary middleware. Users typically are not able to create servers and/or inject middleware.
1. Even more importantly, the execution model on serverless platforms is that the execution may be terminated arbitrarily after the serverless function executes (see [this issue](https://github.com/googleapis/nodejs-logging-bunyan/issues/304)). This means that background work (e.g. draining of log buffers) may not necessarily make it out of the container reliably.

At the moment, the recommended way to log on GCF is to write to the console as that can be reliably transported by the container infrastructure. You can use winston directly with the console logger.

We are evaluating better ways of integrating the middleware with serverless platforms like GCF, but that requires more work before we can have something concrete.

The second issue you raised regarding the hard-coding of the transport addition onto winston is reasonable. I'm going to split that off into a separate issue so that these can be discussed independently: https://github.com/googleapis/nodejs-logging-winston/issues/368

**Comment by jacobg** [comment]
Thanks @ofrobots for explaining how it works. That makes sense. For now, I'll use console logging, and try out the log entry object metadata.

I hope you're able to come up with a solution for GCF, because the request bundling would be awesome! I enjoy that with App Engine. I'll take a look a maybe how I can do that with console logging for now.

**Comment by ofrobots** [comment]
Great. I'm going to close this issue as addressed. Fix for #368 over at #369.

**Comment by techieshark** [comment]
>  For now, I'll use console logging, and try out the log entry object metadata.

@jacobg sorry to bug you here, but did you get this working? I found console.log with a raw object just results in one log entry per console line.

**Comment by jacobg** [comment]
> > For now, I'll use console logging, and try out the log entry object metadata.
> 
> @jacobg sorry to bug you here, but did you get this working? I found console.log with a raw object just results in one log entry per console line.

No I didn't try
